include/deal.II/lac/read_write_vector.templates.h

```diff
@@ -124,6 +124,19 @@ namespace LinearAlgebra
 
 
 
+#ifdef DEAL_II_WITH_CXX11
+  template <typename Number>
+  template <typename Functor>
+  void
+  ReadWriteVector<Number>::apply(const Functor &func)
```

Comment:
The implementation of this function is brief enough that I don't think there would be much harm moving it to the `.h` file. Or are you concerned about having to `#include` whatever header it is that defines `internal::parallel_for`?